ring/lifecycler.go

```diff
@@ -759,11 +779,42 @@ func (i *Lifecycler) compareTokens(fromRing Tokens) bool {
 	return true
 }
 
+func (i *Lifecycler) waitBeforeJoining(ctx context.Context) error {
+	retries := backoff.New(ctx, backoff.Config{
+		MinBackoff: 10 * time.Second,
+		MaxBackoff: 1 * time.Minute,
+		MaxRetries: 5,
```

Comment:
We agreed that timeout should be based on "time" and not "max retries", so MaxRetries should be set to 0. Moreover I think we can be a more aggressive on the speed of retries.

```suggestion
		MinBackoff: 1 * time.Second,
		MaxBackoff: 1 * time.Second,
		MaxRetries: 0,
```